src/TuxArchitectStores.js

```diff
@@ -0,0 +1,108 @@
+  }
+  //return __registerId__ from storeToWaitFor
+  storeRegistrationId = storeToWaitFor.__registerId__;
+  if (storeRegistrationId) {
+    return storeRegistrationId;
+  } else {
+    //if store does not have __registerId__ than it has not been registered to the dispatcher so throw
+    throw new Error('store is waiting for a store that has not been registered to any actions');
+  }
+};
+
+//itNeeds FUNCTION: registers the input(s) and/or store(s) that the store will need to waitFor, returns this architectChain instance to allow for method chaining, updates "and" method to allow chaining via conjuction
+//@param inputOrStore STRING: input string to be checked against the architecture object for the corresponding store [ALTERNATE OBJECT: if a store is passed in than lookup __registerId__ on the passed in store] [ALTERNATE ARRAY: can pass in an array of inputs and/or stores to be registered, method chaining will still function properly in this case]
+ArchitectChain.prototype.itNeeds = function (inputOrStore) {
+  //array of ids to register to the store
+  var waitForIds = [];
+  //if inputOrStore is array
+  if (Array.isArray(inputOrStore)) {
+    var inputOrStoreLength = inputOrStore.length;
+    //loop through inputOrStore array and push register id into waitForIds for each element in array
+    for (var i = 0; i < inputOrStoreLength; i++) {
+      waitForIds.push(this._getRegisterId(inputOrStore[i]));
+    }
+  } else {
+    //if inputOrStore is not an array push its corresponding register id into array
+    waitForIds.push(this._getRegisterId(inputOrStore));
+  }
+  //if the storeToArchitect has already been architected at some point than add waitForIds array on to existing array
+  var tuxArchitecture = this.storeToArchitect.__tuxArchitecture__;
+  if (tuxArchitecture) {
+    tuxArchitecture = tuxArchitecture.concat(waitForIds);
+  } else {
+    //if it has not been architected than just use waitForIds array
+    tuxArchitecture = waitForIds;
+  }
+  //attach architecture array to storeToArchitect at the key of __tuxArchitecture__
+  this.storeToArchitect.__tuxArchitecture__ = tuxArchitecture;
+  //update "and" method so user can chain additional needed inputs with "and"
+  this.and = this.itNeeds;
+  //return this architectChain instance to enable method chaining
+  return this;
+};
+
+//_registerOutputToArchitecture FUNCTION: internal function, registers an output STRING key to the architecture OBJECT with the value of storeToArchitect so that the storeToArchitect can later be looked up by passing the same output STRING to itNeeds
+//@param output STRING: output STRING to which storeToArchitect will be mapped to on the architecture OBJECT
+ArchitectChain.prototype._registerOutputToArchitecture = function (output) {
+  //if this output is already registered to architecture OBJECT, throw error
+  if (architecture[output]) {
+    throw new Error('output: "' + output + '" is already registered to a store');
```

Comment:
Would it make sense to indicate which store the specified output is registered to in this Error?